src/infra/docs/rustc-ci.md

```diff
@@ -63,25 +76,39 @@ Since the merge commit is based on the latest master and only one can be tested
 at the same time, when the results are green master is fast-forwarded to that
 merge commit.
 
+The `auto` branch and other branches used by bors live on a fork of the rust-lang/rust: [rust-lang-ci/rust]. This as originally done to some security limitations in GitHub Actions. These limitations have been addressed, but we've not yet done the work of removing the use of a fork.
+
 Unfortunately testing a single PR at the time, combined with our long CI (~3
-hours for a full run), means we can’t merge too many PRs in a single day, and a
+hours for a full run)[^1], means we can’t merge too many PRs in a single day, and a
 single failure greatly impacts our throughput for the day. The maximum number
 of PRs we can merge in a day is around 8.
 
+The large CI run times and requirement for a large builder pool is worth it because it: 
+* allows perf testing even at a later date 
+* allows bisection when bugs are discover later
+* ensures release quality since if we're always releasing, we catch problems very early
+
+Bors [runs on ecs](https://github.com/rust-lang/simpleinfra/blob/master/terraform/bors/app.tf) and uses a sqlite database running in a volume as storage.
+
+[^1]: As of December 2022, the bottleneck are the macOS builders. Hopefully faster macOS builders will be coming soon!
```

Comment:
Small nit: We recently got larger macOS builders. The current bottleneck, by a very large margin, is dist-x86_64-linux and dist-x86_64-linux-alt (I believe because they are doing BOLT and PGO).